chipsec/chipset.py

```diff
@@ -628,19 +628,41 @@ def init_cfg_bus( self ):
             enum_devices = []
         if QUIET_PCI_ENUM:
             logger().HAL = old_hal_state
+
+        # store entries dev_fun_vid_did = [list of bus entries]
+        for enum_dev in enum_devices:
+            cfg_str = "{:0>2X}_{:0>2X}_{:04X}_{:04X}".format(*enum_dev[1:5])
+            if cfg_str in self.Cfg.BUS.keys():
+                self.Cfg.BUS[cfg_str].append(enum_dev[0])
+            else:
+                self.Cfg.BUS[cfg_str] = [enum_dev[0]]
+
+        # convert entries with matching configuration file names
         for config_device in self.Cfg.CONFIG_PCI:
             device_data = self.Cfg.CONFIG_PCI[config_device]
             xml_vid  = device_data.get( 'vid', None )
             xml_did  = device_data.get( 'did', None )
-            if (xml_vid and xml_did):
-                bus_list = []
-                did_list = [int(_, 16) for _ in xml_did.split(',')]
-                for enum_dev in enum_devices:
-                    if ((int(device_data['dev'], 16), int(device_data['fun'], 16), int(xml_vid, 16)) == enum_dev[1:4]) and (enum_dev[4] in did_list):
-                        bus_list.append( hex(enum_dev[0]) )
-                        if logger().DEBUG: logger().log( '    + {:16s}: VID 0x{:04X} - DID 0x{:04X} -> Bus 0x{:02X}'.format(config_device, enum_dev[3], enum_dev[4], enum_dev[0]) )
-                if len(bus_list):
-                    self.Cfg.BUS[ config_device ] = bus_list
+            # if the vid and did are present within the configuration file attempt to replace generic name with configuration name
+            if xml_vid and xml_did:
+                did_list = []
+                # gather list of device id: device id may have single entry, multiple entries, end in "X", or specified by a range "-"
+                if xml_did:
+                    for tdid in xml_did.split(','):
+                        if tdid[-1].upper() == "X":
+                            for rdv_value in range(tdid, tdid+0xF):
```

Comment:
If `tdid[-1]` is an `'x'`, should it be replaced with 0 and converted to an **int**?   And should the `+0xF` be `+0x10`, assuming x is 0?

Reply:
Looks like a line is missing, will update.